lib/bamboo/mailer.ex

```diff
@@ -131,13 +135,22 @@ defmodule Bamboo.Mailer do
 
     if email.to == [] && email.cc == [] && email.bcc == [] do
       debug_unsent(email)
+      email
     else
       debug_sent(email, adapter)
-      config.deliver_later_strategy.deliver_later(adapter, email, config)
+      response =
+        adapter
+        |> config.deliver_later_strategy.deliver_later(email, config)
+        |> parse_response
+      email |> Map.merge(%{response: response})
     end
-    email
   end
 
+  defp parse_response(%Bamboo.Response{} = response), do: response
+  defp parse_response({:ok, _pid}), do: nil
+  defp parse_response(:noop), do: nil
+  defp parse_response(response), do: raise "Unknown response from adapter: #{inspect response}"
```

Comment:
Currently, a delivery strategy will need to return either `%Response{}`, `{:ok, pid}` or `:noop`. This might be too restrictive. Another option is to make the default case `nil` for anything besides `%Response{}` instead of raising.